### PR TITLE
feat: PR Template & src-docs

### PR DESCRIPTION
**Pull request by yanksyoon**
Applicable spec: N/A

### Overview

This PR adds PR-template and src-docs.

### Rationale

To facilitate PR reviews.

### Juju Events Changes

None.

### Module Changes

None.

### Library Changes

None.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Manging Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [ ] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

There is no documentation update needed.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+### Library Changes
+
+<!-- Any changes to charm libraries -->
+
+### Checklist
+
+- [ ] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
+- [ ] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
+- [ ] The changes are compliant with [ISD054 - Manging Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
```
Comment (cbartz) [inline .github/pull_request_template.md:27]:
```suggestion
- [ ] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Contributing
+
+Build the OCI image:
+
+```bash
+cd jenkins_agent_k8s_rock
+rockcraft
+```
+
+Push the OCI image to microk8s:
```
Comment (cbartz) [inline CONTRIBUTING.md:10]: Maybe add a note that the microk8s registry has to be enabled? In the [Indico Contribute docs](https://charmhub.io/indico/docs/how-to-contribute) its mentioned like this: 

>Note that the microk8s registry needs to be enabled using microk8s enable registry.

File: .github/pull_request_template.md (hunk 1 above)
Comment (yanksyoon) [reply .github/pull_request_template.md]: Ach blind copy paste, fixing the source. Thanks for the catch!

File: CONTRIBUTING.md (hunk 2 above)
Comment (yanksyoon) [reply CONTRIBUTING.md:10]: Updated, thank you!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+(Note that the microk8s registry needs to be enabled using microk8s enable registry.)
```
Comment (cbartz) [inline CONTRIBUTING.md:11]:
```suggestion
(Note that the microk8s registry needs to be enabled using `microk8s enable registry`.)
```
A bit nitpicky, see that as optional

**Review by cbartz — APPROVED** [review]

**Review by merkata — DISMISSED** [review]

**Review by cbartz — APPROVED** [review]

**Review by arturo-seijas — APPROVED** [review]
LGTM. Next time, consider splitting unrelated changes in multiple PRs

**Pull request merged**